server/controllers/userProfileController.js

```diff
@@ -0,0 +1,57 @@
+const User = require('../models/User')
+const UserProfile = require('../models/UserProfile')
+const asyncHandler = require('express-async-handler')
+
+
+exports.createUserProfile = asyncHandler(async (req, res) => {
+  try {
+    const profile = await new UserProfile(req.body)
+    const userProfile = await User.updateOne(
+      { _id: req.params.userId },
+      {
+        $set: {
+          profile
+        }
```

Comment:
You don't need to update the profile like this, You could try an easier version something similar to this
 ```

 const { userId, userType, email } = req.body;

  const profile = await Profile.create({
    userId,
    email,
    userType,
  });
```